### ISSUE TITLE
Support title from xterm.js

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#22325]
From @pharmmd-rich in https://github.com/Microsoft/vscode/issues/22319

> The dropdown list currently displays a list of processes, however, I like giving my terminals custom names in OS-X. The drop down list does not appear to honor these names. (they are set using: printf '\033]0;title\007'".

Open to PRs, we will want to handle `xterm.on('title', listener)` inside `TerminalInstance` and use that over the pty name if it's set. Currently I'm assuming this will only fire if the user has set up a title they want to use, whether of not the event is guarenteed to fire on new terminals might change my thinking on how it works.

**Comment by Tyriar** [comment]
@Lixire here's one of the PRs with some extra detail in what needs to happen https://github.com/Microsoft/vscode/pull/22454#issuecomment-285905297

**Comment by Tyriar** [comment]
Just added a new setting for this, set:

```
"terminal.integrated.experimentalUseTitleEvent": true
```

To use the title from the VT sequence.

**Comment by llibicpep** [comment]
@Tyriar thanks a lot! I just tried it, and it seems that it works, with a little issue though.
So I usually right click on a sub-folder and choose `Open in Terminal`. When I do that with `experimentalUseTitleEvent` enabled, the title automatically set to `1: /bin/bash`. But if I `cd` anywhere and all subsequent `cd`s changing the title to the `cwd`.

**Comment by Tyriar** [comment]
Sounds like the first event isn't getting fired, should get fixed before we remove experimental.

**Comment by cr4ckm3** [comment]
Also it shows whole path with user@userdomain, would be perfect if I can choose to show only last folder name or set in workspace config the name of the terminal. Thanks a lot!

**Comment by llibicpep** [comment]
I've been using vscode on Mac OS Mojave with bash and it seems to be working as @cr4ckm3 described (dropdown name is an equivalent of `basename $(pwd)`). I just upgraded to Catalina where they made `zsh` a default shell and I decided to give it a shot. With `zsh` it uses `${whoami}@${hostname} $(print -rD $PWD)`,  which is too long and the only visible part is username+hostname so I couldn't tell which terminal I'm looking at right away without clicking on dropdown to see expanded list.
